### ISSUE TITLE
[Pods 2.8 Bug] Single Dropdown & Radio relationship fields not saving

### ISSUE THREAD

**Issue by pdclark** [pods-framework/pods#6131]
## Pods 2.8 version

`2.8.0-rc-1`

## Description

Single-select Dropdown and Radio relationship fields appear to not be saving.

For drop-down
* The database appears to get an incorrect ID saved to the meta.
* UI for field shows blank after reload.

For radio
* Database appears to save correct value.
* UI for field shows blank after reload.

## Testing instructions

Package export:
```json
{
    "meta": {
        "version": "2.8.0-rc-1",
        "build": 1631310610
    },
    "pods": [
        {
            "name": "chapter",
            "id": 4,
            "label": "Chapters",
            "description": "",
            "type": "post_type",
            "storage": "meta",
            "label_singular": "Chapter",
            "public": "1",
            "show_ui": "1",
            "rest_enable": "1",
            "supports_title": "1",
            "supports_editor": "1",
            "publicly_queryable": "1",
            "exclude_from_search": "0",
            "capability_type": "post",
            "capability_type_custom": "chapter",
            "capability_type_extra": "1",
            "has_archive": "0",
            "hierarchical": "0",
            "rewrite": "1",
            "rewrite_with_front": "1",
            "rewrite_feeds": "0",
            "rewrite_pages": "1",
            "query_var": "1",
            "can_export": "1",
            "default_status": "draft",
            "supports_author": "0",
            "supports_thumbnail": "0",
            "supports_excerpt": "0",
            "supports_trackbacks": "0",
            "supports_custom_fields": "0",
            "supports_comments": "0",
            "supports_revisions": "0",
            "supports_page_attributes": "0",
            "supports_post_formats": "0",
            "built_in_taxonomies_category": "0",
            "built_in_taxonomies_link_category": "0",
            "built_in_taxonomies_post_tag": "0",
            "built_in_taxonomies_wp_theme": "0",
            "show_in_menu": "1",
            "menu_position": "0",
            "show_in_nav_menus": "1",
            "show_in_admin_bar": "1",
            "pfat_enable": "0",
            "pfat_append_single": "append",
            "pfat_filter_single": "the_content",
            "pfat_append_archive": "append",
            "pfat_filter_archive": "the_excerpt",
            "pfat_run_outside_loop": "0",
            "read_all": "0",
            "write_all": "0",
            "groups": [
                {
                    "name": "more_fields",
                    "id": 5,
                    "label": "More Fields",
                    "description": "",
                    "weight": 0,
                    "fields": [
                        {
                            "name": "chapter_course",
                            "id": 8,
                            "label": "Chapter Course",
                            "description": "",
                            "weight": 0,
                            "type": "pick",
                            "pick_object": "post_type",
                            "pick_val": "course",
                            "post_status": "publish",
                            "required": "0",
                            "pick_format_type": "single",
                            "pick_format_single": "dropdown",
                            "pick_format_multi": "checkbox",
                            "pick_display_format_multi": "default",
                            "pick_display_format_separator": ", ",
                            "pick_allow_add_new": "1",
                            "pick_taggable": "0",
                            "pick_show_icon": "1",
                            "pick_show_edit_link": "1",
                            "pick_show_view_link": "1",
                            "pick_limit": "0",
                            "pick_user_role": "Administrator",
                            "roles_allowed": "administrator",
                            "rest_pick_response": "array",
                            "rest_pick_depth": "1",
                            "unique": "0",
                            "admin_only": "0",
                            "restrict_role": "0",
                            "restrict_capability": "0",
                            "hidden": "0",
                            "read_only": "0",
                            "rest_read": "0",
                            "rest_write": "0",
                            "sister_id": "-- Select One --",
                            "pick_table": "-- Select Table --"
                        }
                    ]
                }
            ]
        },
        {
            "name": "course",
            "id": 6,
            "label": "Courses",
            "description": "",
            "type": "post_type",
            "storage": "meta",
            "label_singular": "Course",
            "public": "1",
            "show_ui": "1",
            "rest_enable": "1",
            "supports_title": "1",
            "supports_editor": "1",
            "groups": [
                {
                    "name": "more_fields",
                    "id": 7,
                    "label": "More Fields",
                    "description": "",
                    "weight": 0,
                    "fields": []
                }
            ]
        }
    ]
}
```

- Create a CPT-to-CPT single-select relationship field with Drop-down or Radio as the UI, or import the JSON above.
- Try to modify and save the relationship.
- Reload the edit interface.
- See the field did not save.
- Check the database — see the database value might also be incorrect.

## Screenshots / screencast

https://pods.pd.cm/wp-content/mu-plugins/ominem/relationship-dropdown-radio.mp4
(Video doesn't play for me in-browser, but does play if downloaded.)

## Possible workaround

* `2.7.30` works okay.
* Auto-complete and List views work okay.

## Other super helpful information

There are 12 times more trees on Earth than stars in the Milky Way.

**Comment by zrothauser** [comment]
@sc0ttkclark The first part of this (with the single dropdown) looks like an issue with the fieldItemData that is provided for Select fields when the pick_val is a CPT. Instead of the `id` attribute being a post ID, it's an index starting at 0 after the placeholder. That's the value that then gets saved. 

```
"fieldItemData": [
    {
        "id": "",
        "icon": "dashicons-admin-post",
        "name": "-- Select One --",
        "edit_link": "http://pods.local/wp-admin/post.php?post=1985&action=edit&pods_modal=1",
        "link": "http://pods.local/another-post/",
        "selected": false
    },
    {
        "id": "0",
        "icon": "dashicons-admin-post",
        "name": "(No Title)",
        "edit_link": "http://pods.local/wp-admin/post.php?post=1985&action=edit&pods_modal=1",
        "link": "http://pods.local/another-post/",
        "selected": false
    },
    {
        "id": "1",
        "icon": "dashicons-admin-post",
        "name": "Test Post",
        "edit_link": "http://pods.local/wp-admin/post.php?post=1&action=edit&pods_modal=1",
        "link": "http://pods.local/hello-world/",
        "selected": false
    }
],
```

The second part of this (with the radio buttons) is a typing issue, I'll have a PR to address this.

**Comment by sc0ttkclark** [comment]
I will fix the fieldItemData id issue @zrothauser, thanks!

**Comment by sc0ttkclark** [comment]
Fixed the fieldItemData id issue via e9b508f0c3a66b08a876a11818a4235552a07163

**Comment by sc0ttkclark** [comment]
Fixed via https://github.com/pods-framework/pods/pull/6140
